### ISSUE TITLE
Yarn in Jenkins CI - build console looks bad

### ISSUE THREAD

**Issue by osher** [yarnpkg/yarn#8444]
### Bug description

**Command**

```sh
yarn install --network-timeout 180000 --non-interactive --no-progress
```

**What is the current behavior?**

Between each two lines `yarn` has to say, the build console (aka build log) is full unexpected number of:
```
[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G[0K[1G
```
It does seem that the number of `[1G[0K` is somehow proportional to the time waited between the two logged lines.

**What is the expected behavior?**

I want just the text. I did say `--no-progress`, and `--non-interactive` and hoped it will go away, but alas, no.
If it's only just a switch I'm missing....

**Steps to Reproduce**

1. The command runs in Jenkins, within a docker with the project mounted as it's work dir.

**Environment**

- Node Version: `12.9.0
- Yarn v1 Version: `1.15.1` and also `1.17.1`
- OS and version: `alpine 3.9`

**Comment by osher** [comment]
We spoke about it here, and thought maybe this info will be helpful:

I don't know if it matters, but since we want visibility into build steps, we're not just running a docker build, so that `yarn install` is not a command inside a Dockerfile, but a command ran on a container. Let me explain:


Before building, first - for testing - we run a node container and `exec` all the commands on it - i.e - use a jenkins pipeline `stage (name){ .. }` for each of the required steps:
 - launch container
 - install dependencies
 - test
 - lint
 - merge cover reports

where, launch container - means runing a container with a keep alive CMD in the spirit of `while (true) sleep 600`
and each step after it - runs from inside the `stage(name) { ... }` a shell command in the spirit of `docker exec <container-id> <command>`, where in our case - `<command>` is the `yarn install...`.
The motive is that a result, we have a UI indication for each of these steps.

However, I'm afraid that `yarn` does not understand it's running in non TTL mode, and doing this spinning thing.

I hope there is a way to ask it not to...

I hope this helps.

**Comment by osher** [comment]
I found the culprit and have a work-around.
Since blue-ocean presents things in colors - then the `docker exec` was implemented with `-t`.

All other tools we use implement a flag that tells them explicitly what mode to run, yarn does not.

Since the most of our projects uses npm, we had to implement a custom settings that asks for yarn, and with it - a custom settings that asks that the install stage will not use `-it`.

It would have been much better if yarn had a switch to disable ttl mode explicitly.
